crypto/o_str.c

```diff
@@ -53,7 +54,7 @@ void *CRYPTO_memdup(const void *data, size_t siz, const char* file, int line)
         return NULL;
 
     ret = CRYPTO_malloc(siz, file, line);
-    if (ret == NULL) {
+    if (ret == NULL && errno == ENOMEM) {
```

Comment:
Calling memcpy with a NULL value for `ret` is (I think) undefined behaviour. So this change does not look correct to me. Possibly it is better to additionally confirm that `siz` is > 0 in the line `if (data == NULL || siz >= INT_MAX)` above. Regardless of whether `malloc` has failed with ENOMEM, `CRYPTO_malloc` has failed if it gives a NULL response and it seems to me that an `ERR_R_MALLOC_FAILURE` error is appropriate.

Reply:
@mattcaswell Thanks for the suggestion but I removed the change in `CRYPTO_memdup()` because https://github.com/openssl/openssl/pull/17507#issuecomment-1013064386 suggested that the change should be made from the caller's side.